flexmeasures/conftest.py

```diff
@@ -156,6 +162,23 @@ def create_test_accounts(db) -> dict[str, Account]:
     )
 
 
+def add_consult_account_access_to_prosumer_account(db, test_accounts):
+    account = Account.query.filter_by(name="Test Prosumer Account").one_or_none()
+    consultant_account_role = AccountRole(
+        name="Consultant",
+        description="Account that prosumer can access",
```

Comment:
This is the wrong way around. The consultant can access accounts of its clients (which in your case here is the Prosumer). Please reverse this and your test.

Reply:
Changed the `conftest` by adding a ConsultantClient account with an asset. The Consultant account has access to the ConsultantClient account.